rbi/annotations/activesupport.rbi

```diff
@@ -259,3 +265,134 @@ class Time
   sig { returns(TrueClass) }
   def present?; end
 end
+
+class String
+  sig { returns(TrueClass) }
+  def acts_like_string?; end
+
+  sig { params(position: T.any(Integer, Regexp, String, T::Range[Integer])).returns(T.nilable(T.self_type)) }
+  def at(position); end
+
+  sig { params(first_letter: Symbol).returns(T.self_type) }
+  def camelcase(first_letter = :upper); end
+
+  sig { params(first_letter: Symbol).returns(T.self_type) }
+  def camelize(first_letter = :upper); end
+
+  sig { returns(T.self_type) }
+  def classify; end
+
+  sig { returns(T.nilable(Module)) }
+  def constantize; end
+
+  sig { returns(T.self_type) }
+  def dasherize; end
+
+  sig { returns(T.self_type) }
+  def deconstantize; end
+
+  sig { returns(T.self_type) }
+  def demodulize; end
+
+  sig { params(string: String).returns(T::Boolean) }
+  def exclude?(string); end
+
+  sig { params(limit: Integer).returns(T.self_type) }
+  def first(limit = 1); end
+
+  sig { params(separate_class_name_and_id_with_underscore: T.untyped).returns(T.self_type) }
+  def foreign_key(separate_class_name_and_id_with_underscore = true); end
+
+  sig { params(position: Integer).returns(T.self_type) }
+  def from(position); end
+
+  sig { returns(ActiveSupport::SafeBuffer) }
+  def html_safe; end
+
+  sig { params(capitalize: T.untyped, keep_id_suffix: T.untyped).returns(T.self_type) }
+  def humanize(capitalize: true, keep_id_suffix: false); end
```

Comment:
Always a tough balance to strike when adding types to untyped code :)

I opted to err on the side of flexibility here since any truthy value [will work here](https://github.com/rails/rails/blob/488a7ce188803828c3e7111e23e70108d761c283/activesupport/lib/active_support/inflector/methods.rb#L151) but happy to change to boolean if you want a more strict type.

Reply:
I'm not familiar with its usages so if you think being flexible is good for cases where an object is supplied I'm okay with it.